test/connectObservable.test.tsx

```diff
@@ -35,6 +35,49 @@ describe("connectObservable", () => {
     expect(result.current).toEqual(1)
   })
 
+  it("updates with the last emitted value", async () => {
+    const numberStream = new BehaviorSubject(1)
+    const [useNumber] = connectObservable(numberStream)
+    const { result } = renderHook(() => useNumber())
+    expect(result.current).toBe(1)
+
+    act(() => {
+      numberStream.next(2)
+    })
+    expect(result.current).toBe(2)
+  })
+
+  it("allows React to batch synchronous updates", async () => {
```

Comment:
This is the closest I could to try any batching stuff - What this is doing is by testing a very simple case: updates happening in a callback controlled by React (e.g. events) are getting batched.

I wonder if this test actually adds any value. I don't know the specifics of `act`, but if it's calling `unstable_batchUpdates` internally, then the value of this test is basically that updates are being dispatched in the same event loop.